sway-lsp/src/traverse/parsed_tree.rs

```diff
@@ -779,58 +778,49 @@ impl Parse for ParsedDeclId<EnumDeclaration> {
     }
 }
 
-impl Parse for ParsedDeclId<ImplTrait> {
+impl Parse for ParsedDeclId<ImplSelfOrTrait> {
     fn parse(&self, ctx: &ParseContext) {
-        let impl_trait = ctx.engines.pe().get_impl_trait(self);
-        adaptive_iter(&impl_trait.trait_name.prefixes, |ident| {
+        let impl_self_or_trait = ctx.engines.pe().get_impl_self_or_trait(self);
+        adaptive_iter(&impl_self_or_trait.trait_name.prefixes, |ident| {
             ctx.tokens.insert(
                 ctx.ident(ident),
                 Token::from_parsed(AstToken::Ident(ident.clone()), SymbolKind::Module),
             );
         });
         ctx.tokens.insert(
-            ctx.ident(&impl_trait.trait_name.suffix),
+            ctx.ident(&impl_self_or_trait.trait_name.suffix),
             Token::from_parsed(
-                AstToken::Declaration(Declaration::ImplTrait(*self)),
+                AstToken::Declaration(Declaration::ImplSelfOrTrait(*self)),
                 SymbolKind::Trait,
             ),
         );
-        impl_trait.implementing_for.parse(ctx);
-        adaptive_iter(&impl_trait.impl_type_parameters, |type_param| {
-            type_param.parse(ctx);
-        });
-        adaptive_iter(&impl_trait.items, |item| match item {
-            ImplItem::Fn(fn_decl) => fn_decl.parse(ctx),
-            ImplItem::Constant(const_decl) => const_decl.parse(ctx),
-            ImplItem::Type(type_decl) => type_decl.parse(ctx),
-        });
```

Comment:
We probably want to keep parsing these fields even with this change unless i'm missing something?

Reply:
True, thanks for pointing that out, was getting confused.